### PR TITLE
support standard touch scrolling on iOS for main content

### PR DESCRIPTION
**Pull request by sqs**
Previously, the scrollable parts of the web app (e.g., the blob, panel, and file tree; or the page for non-blob repository pages) would scroll "stiffly" on iOS. They would not have the same WebKit overflow "bounce" scrolling behavior that users expect on iOS. This fixes it so that the scrollable areas do behave as users expect on iOS.

See https://github.com/sourcegraph/sourcegraph/pull/1975#discussion_r255170228 for context.

### CODE REVIEW

**Comment by sqs** [comment]
@felixfbecker 	All of these `height:100%`s are necessary. The reason why, I think, is that they force the ancestor elements to be the window height, which ensures that the overflowing content scrolls instead of gets hidden.

There are a few remaining problems with scrolling, but they are rare:

- If the window height changes while you are scrolling (which happens sometimes when iOS Safari shows or hides the address bar when you scroll down), then you might not be able to scroll all of the way down.
- If you touch-scroll on a fixed part of the UI, it will try to scroll the whole viewport, not just the scrollable area. iOS then imposes a 1500msec delay until it lets you scroll another scrollable area.

Neither of these are blockers IMO because they are rare and they are not regressions.

**Comment by felixfbecker** [comment]
Is the `overflow: hidden` needed too?

**Comment by sqs** [comment]
Yes. Try it out yourself and see, to review it.

**Comment by felixfbecker** [comment]
Removing `overflow: hidden` does not make any difference for me:

ezgif-4-89db9db9a955

**Comment by sqs** [comment]
@felixfbecker:

> Removing `overflow: hidden` does not make any difference for me:

Indeed, you are right. Thanks for the catch! I removed it.

File: web/src/Layout.scss [hunk 1]
```diff
@@ -1,6 +1,8 @@
 .layout {
     display: flex;
     width: 100%;
+    height: 100%;
+    overflow: hidden;
```
Comment (felixfbecker) [inline web/src/Layout.scss]: remove here too?

Comment (sqs) [reply web/src/Layout.scss]: thx, done

**Pull request merged**